app/configurator/components/ui-helpers.ts

```diff
@@ -368,16 +369,29 @@ export const extractDataPickerOptionsFromDimension = ({
   const noneLabel = "None";
 
   if (values.length) {
-    const options = values.map((d) => {
-      return {
-        label: `${d.value}`,
-        value: `${d.value}`,
-      };
-    });
+    const [minValue, maxValue] = isTemporalDimension(dimension)
+      ? [values[0].value as string, values[values.length - 1].value as string]
+      : [
+          values[0].position as string,
+          values[values.length - 1].position as string,
+        ];
+    const options = isTemporalDimension(dimension)
+      ? values.map((d) => {
+          return {
+            label: `${d.value}`,
+            value: `${d.value}`,
+          };
+        })
+      : values.map((d) => {
+          return {
+            label: `${d.label}`,
+            value: `${d.position}`,
+          };
+        });
```

Comment:
Nit: Wonder if we should use ts-pattern here for exhaustive check.

Reply:
Yup 👍